src/Common/Commands.Common.Authentication.Test/AuthenticationFactoryTests.cs

```diff
@@ -135,6 +137,50 @@ public void CanAuthenticateWithAccessToken()
             VerifyToken(checkKVToken, kvToken, userId, tenant);
         }
 
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void CanAuthenticateUsingMSIDefault()
+        {
+            AzureSessionInitializer.InitializeAzureSession();
+            string expectedAccessToken = Guid.NewGuid().ToString();
+            string expectedToken2 = Guid.NewGuid().ToString();
+            string tenant = Guid.NewGuid().ToString();
+            string userId = "user1@contoso.org";
+            var account = new AzureAccount
+            {
+                Id = userId,
+                Type = AzureAccount.AccountType.ManagedService
+            }; ;
+            var environment = AzureEnvironment.PublicEnvironments["AzureCloud"];
+            string expectedResource = environment.ActiveDirectoryServiceEndpointResourceId;
+            var builder = new UriBuilder(AuthenticationFactory.DefaultMSILoginUri);
+            builder.Query = string.Format("resource={0}", Uri.EscapeDataString(environment.ActiveDirectoryServiceEndpointResourceId));
+            var defaultUri = builder.Uri.ToString();
+
+            IDictionary<string, ManagedServiceTokenInfo> responses = new Dictionary<string, ManagedServiceTokenInfo>(StringComparer.OrdinalIgnoreCase)
```

Comment:
What is our guidance on explicit vs. inferred types?

Reply:
Should use type inference where it is possible.  In this case, because the test class should be able to handle any usage for any type, it is required to explicitly type the input dictionary.

Reply:
I mean in the declaration.  It doesn't really need to be the interface type, correct?